docs/requirements/channel-upgradability-requirements.md

```diff
@@ -0,0 +1,224 @@
+
+## Objectives
+
+To enable existing channels to upgrade the application module claiming the channel or add middleware to both ends of an existing channel whilst retaining the state of the channel. 
+
+## Scope
+
+A new `ChannelEnd` interface is defined after a channel upgrade, the scope of these upgrades is detailed in the table below.
+
+| Features  | Release |
+| --------- | ------- |
+| Performing a channel upgrade results in an application module changing from v1 to v2, claiming the same `channelID` and `portID` | v1 |
+| Performing a channel upgrade results in a channel with the same `channelID` and `portID` changing the channel ordering from a higher to lower degree of ordering | v1 |
+| Performing a channel upgrade results in a channel with the same `channelID` and `portID` having additional middleware added to the application stack on both sides of the channel | v1 |
+| Performing a channel upgrade results in a channel with the same `channelID` and `portID` modifying the `connectionHops` | v1 |
+
+# User requirements
+
+## Use cases
+
+Sample use cases:
+
+- Upgrading an existing application module from v1 to v2, e.g. new features could be added to the existing ICS 20 application module which would result in a new version of the module.
+- Adding middleware on both sides of an existing channel, e.g. relayer incentivisation middleware, ICS 29, requires middleware to be added to both ends of a channel to incentivise the `recvPacket`, `acknowledgePacket` and `timeoutPacket`.
+
+Upgrades can be initiated in two possible ways:
+
+- __Permissionless__: upgrade initiation is approved by governance or group proposal and a relayer can submit `MsgChannelUpgradeInit`.
+- __Permissioned__: upgrade initiation is approved by governance or group proposal and `MsgChannelUpgradeInit` is automatically submitted on proposal passing.
+
+The upgrade approval process can be represented in terms of the following parameters:
+
+- The set of port ID, channel ID pairs that specify the channels that can be upgraded. It can be possible to specifit a single channel or a set of channels satisfiying certain port ID, channel ID conditions. Examples: upgrading can be approved for __a)__ channels with port ID, channel ID combinations (`transfer`, `channel-0`) & (`transfer`, `channel-1`), __b)__ all channels for port ID `transfer`, __c)__ all channels with port ID starting with `icacontroller` (so that all ICA channels on a controller chain, regardless of the owner address appended to the port ID, can be upgraded).
```

Comment:
```suggestion
- The set of port ID, channel ID pairs that specify the channels that can be upgraded. It can be possible to specify a single channel or a set of channels satisfying certain port ID, channel ID conditions. Examples: upgrading can be approved for __a)__ channels with port ID, channel ID combinations (`transfer`, `channel-0`) & (`transfer`, `channel-1`), __b)__ all channels for port ID `transfer`, __c)__ all channels with port ID starting with `icacontroller` (so that all ICA channels on a controller chain, regardless of the owner address appended to the port ID, can be upgraded).
```